lib/shark_on_lambda.rb

```diff
@@ -87,6 +87,15 @@ def reset_configuration
       true
     end
 
+    # TODO this should probably be somewhere else
+    def run(event:, context:)
+      adapter = RackAdapters::ApiGateway.new(context: context, event: event)
+      env = adapter.env
+
+      status, headers, body = application.call(env)
+      adapter.build_response(status, headers, body)
+    end
```

Comment:
I really don't like this here, but ApiGatewayHandler is already taken. 🤷‍♂ 
Anyway, these 6 lines are really all it needs.

Reply:
Maybe something like
```
module SharkOnLambda
  class Application 
    def self.run; ....; end
   end
end
```

```
class MyApplication < SharkOnLambda::Application
end
```

```
# serverless.yml
functions:
  api-v2:
    handler: app/api_gateway.MyApplication.run
```

Reply:
Alternatives to `Application` which seems to be kind of private are
* `ApiGatewayHandler.run`
* `Handler.api_gateway`
* `ApiGateway.handle`
* ...